pkg/util/kernel/fs.go

```diff
@@ -31,29 +30,35 @@ func ParseMountInfoFile(pid int32) ([]*mountinfo.Info, error) {
 
 // GetMountPoint returns the mount point of the given path
 func GetMountPoint(path string) (*mountinfo.Info, error) {
-	mi, err := ParseMountInfoFile(int32(os.Getpid()))
+	if path == "" {
+		return nil, fmt.Errorf("empty path")
+	}
+
+	mi, err := mountinfo.GetMounts(nil)
 	if err != nil {
 		return nil, err
 	}
 
-	for path != "" {
+	for {
 		for _, m := range mi {
 			if path == m.Mountpoint {
 				return m, nil
 			}
 		}
 
+		if path == "/" {
```

Comment:
IDK if this is a case that needs to be considered, but if a relative path is ever passed in (`./foo/bar`), I think this will infinite loop.

filepath.Dir(path) will return `.`  if the path is empty or `.`.  `./foo/bar` returns `./foo`, which then returns `.`

Reply:
We can address that in 7.33, but isn't necessary for this backport.